lib/webpacker/file_loader.rb

```diff
@@ -4,21 +4,38 @@ class NotFoundError < StandardError; end
   class FileLoaderError < StandardError; end
 
   class_attribute :instance
-  attr_accessor :data
+  attr_accessor :data, :mtime, :path
 
   class << self
-    def load(path = file_path)
-      self.instance = new(path)
+    def load_instance(path = file_path)
+      if instance.nil? || !production_env? || !file_cached?(path)
+        self.instance = new(path)
+      end
     end
+
+    def file_path
+      raise FileLoaderError.new("Subclass of Webpacker::FileLoader should override this method")
+    end
+
+    private
+      def file_cached?(path)
+        File.exist?(path) && self.instance.mtime == File.mtime(path)
+      end
+
+      # Prefer the NODE_ENV to the rails env.
+      def production_env?
```

Comment:
`Webpacker::Env` already does this - https://github.com/shakacode/webpacker/blob/4cf3fcab5c03203216be7ab0b7f161b591908174/lib/webpacker/env.rb#L19

so if we just call `Webpacker.env.production?` it would be same thing as this